### PR TITLE
Merchant dashboard statistics#29

### LINKED ISSUE #29: User Story 40, Merchant Dashboard Statistics
```
As a merchant
When I visit my dashboard, I see an area with statistics:
- top 5 items I have sold by quantity
- total quantity of items I've sold, and as a percentage against my sold units plus remaining inventory (eg, if I have sold 1,000 things and still have 9,000 things in inventory, the message would say something like "Sold 1,000 items, which is 10% of your total inventory")
- top 3 states where my items were shipped
- top 3 city/state where my items were shipped (Springfield, MI should not be grouped with Springfield, CO)
- name of the user with the most orders from me (pick one if there's a tie)
- name of the user who bought the most total items from me (pick one if there's a tie)
- top 3 users who have spent the most money on my items
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #32

depends on #31

### PR DESCRIPTION
**Pull request by asmolentzov**
Completes User Story 40, Merchant Dashboard Statistics
- Worked on together with Will
- Adds tests and methods for statistics for a particular merchant:
- top 5 items I have sold by quantity
- total quantity of items I've sold, and as a percentage against my sold units plus remaining inventory (eg, if I have sold 1,000 things and still have 9,000 things in inventory, the message would say something like "Sold 1,000 items, which is 10% of your total inventory")
- top 3 states where my items were shipped
- top 3 city/state where my items were shipped (Springfield, MI should not be grouped with Springfield, CO)
- name of the user with the most orders from me (pick one if there's a tie)
- name of the user who bought the most total items from me (pick one if there's a tie)
- top 3 users who have spent the most money on my items
- Updates dashboard show page with statistics
- All tests passing
Closes #29

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -70,4 +70,99 @@ def self.top_cities
...
+    .select("users.*, SUM(order_items.quantity * order_items.order_price) AS units_purchased")
+    .order("units_purchased DESC")
+    .limit(3)
+  end
+  
+  def merchant_order_ids
+    self.items.joins(:orders)
+    .where("orders.status = ?", 1)
+    .select("orders.*").map(&:id)
+  end
```
Comment (saleratus) [inline app/models/user.rb]: Very clever approach. You rule, Anna!!
